src/DeviceHardDriveType.php

```diff
@@ -0,0 +1,106 @@
+ *
+ * http://glpi-project.org
+ *
+ * @copyright 2015-2023 Teclib' and contributors.
+ * @copyright 2003-2014 by the INDEPNET Development Team.
+ * @licence   https://www.gnu.org/licenses/gpl-3.0.html
+ *
+ * ---------------------------------------------------------------------
+ *
+ * LICENSE
+ *
+ * This file is part of GLPI.
+ *
+ * This program is free software: you can redistribute it and/or modify
+ * it under the terms of the GNU General Public License as published by
+ * the Free Software Foundation, either version 3 of the License, or
+ * (at your option) any later version.
+ *
+ * This program is distributed in the hope that it will be useful,
+ * but WITHOUT ANY WARRANTY; without even the implied warranty of
+ * MERCHANTABILITY or FITNESS FOR A PARTICULAR PURPOSE.  See the
+ * GNU General Public License for more details.
+ *
+ * You should have received a copy of the GNU General Public License
+ * along with this program.  If not, see <https://www.gnu.org/licenses/>.
+ *
+ * ---------------------------------------------------------------------
+ */
+
+class DeviceHardDriveType extends CommonDeviceType
+{
+    public $additional_fields_for_dictionnary = ['manufacturer'];
+
+    public static function getTypeName($nb = 0)
+    {
+        return _n('Device hard drive type', 'Device hard drive types', $nb);
+    }
+
+    /**
+     * @since 10.1
+     *
+     * @param string $itemtype
+     * @param HTMLTableBase $base
+     * @param HTMLTableSuperHeader|null $super
+     * @param HTMLTableHeader|null $father
+     * @param array $options
+     *
+     * @return void
+     **/
+    public static function getHTMLTableHeader(
+        string $itemtype,
+        HTMLTableBase $base,
+        ?HTMLTableSuperHeader $super = null,
+        ?HTMLTableHeader $father = null,
+        array $options = []
+    ): void {
+
+        $column_name = __CLASS__;
+
+        if (isset($options['dont_display'][$column_name])) {
+            return;
+        }
+
+        $base->addHeader($column_name, _n('Type', 'Types', 1), $super, $father);
+    }
+
+
+    /**
+     * @since 10.1
+     *
+     * @param HTMLTableRow|null $row
+     * @param CommonDBTM|null $item
+     * @param HTMLTableCell|null $father
+     * @param $options   array
+     *
+     * @return void
+     **/
+    public static function getHTMLTableCellsForItem(
+        ?HTMLTableRow $row = null,
+        ?CommonDBTM $item = null,
+        ?HTMLTableCell $father = null,
+        array $options = []
+    ): void {
+        $column_name = __CLASS__;
+
+        if (isset($options['dont_display'][$column_name])) {
+            return;
+        }
+
+        if ($item->fields["deviceharddrivetypes_id"]) {
+            $row->addCell(
+                $row->getHeaderByName($column_name),
+                Dropdown::getDropdownName(
+                    "glpi_deviceharddrivetypes",
+                    $item->fields["deviceharddrivetypes_id"]
+                )
+            );
+        }
+    }
```

Comment:
I think it would be useless if we handle this like it is done in `DeviceMemory` (see my previous comments).
```suggestion
```